### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by igiloh-pinecone**
These are some test fixes originating from https://github.com/pinecone-io/client_sdk/pull/66/ that never got merged.

I've separated the new multi-threading and GRPC connection tests to a separate PR.

### CODE REVIEW

**Review by rajat08 — APPROVED** [review]

File: tests/unit/test_data_plane.py [hunk 1]
```diff
@@ -28,6 +28,11 @@
 )
 
 def get_random_metadata():
+    # return a random value for metadata key
```
Comment (jhamon) [inline tests/unit/test_data_plane.py:31]: Adding randomness in tests can lead to flakey failures where things fail sometimes but not always, pass on re-run, etc. Can we just hardcode a metadata key into the test using this? If we care about all these options, we should test each of them.

File: tests/unit/test_data_plane.py [hunk 2]
```diff
@@ -418,26 +428,27 @@ def test_query_simple_with_values_mixed_metadata(test_data_plane_index):
...
+        values=[0.1] * vector_dim,
+        namespace=namespace,
+        top_k=10,
+        include_values=True,
+        include_metadata=True,
+        filter={'genre': {'$in': ['action']}}
+    )
+    logger.debug('got query (with filter, with data, with metadata) response: {}', api_response)
+    if not api_response:
+        pytest.skip('no vectors match the filter')
```
Comment (jhamon) [inline tests/unit/test_data_plane.py:447]: I think we should retry or fail if the `api_response` is not what we expect.

File: tests/utils/utils.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import random
 from .remote_index import RemoteIndex
 from _pytest.python_api import approx
+import numpy as np
```
Comment (jhamon) [inline tests/utils/utils.py:11]: Maybe we don't need this if you can refactor our the randomness above.

**Review by jhamon — APPROVED** [review]
I there are still some improvements I'd like to see here to try to make the tests more deterministic, but I'll leave it up to you whether to fix now or merge and fix as a follow-up.

File: tests/unit/test_control_plane.py [hunk 4]
```diff
@@ -3,14 +3,14 @@
...
-from utils.remote_index import PodType
+import numpy as np 
+from ..utils.utils import retry_assert
+from ..utils.remote_index import PodType
 
 env = os.getenv('PINECONE_REGION')
 key = os.getenv('PINECONE_API_KEY')
 client = Client(key, env)
-INDEX_NAME_PREFIX = 'control-plane'
+INDEX_NAME_PREFIX = 'control-plane-' + str(np.random.randint(10000))
```
[the remark this answers is not in the document]
Comment (igiloh-pinecone) [reply tests/unit/test_control_plane.py:13]: The indexes are deleted, see `RemoteIndex.__exit__()`.

The reason for the random index number is to avoid to parallel tests (e.g. from two different PRs \ pushes) using the same index and interfering with each other. Especially _because_ each test deletes its on indexes upon exit.

File: tests/unit/test_data_plane.py (hunk 1 above)
Comment (igiloh-pinecone) [reply tests/unit/test_data_plane.py:31]: Adding randomness also increases the chance of catching some weird edge cases that no one thought of (e.g. updating a vector that had some specific combination of metadata keys, but with some different combination in the update).

If we get to the state which tests are unstable (which currently doesn't seem to be the case), I agree we can decrease randomness.

File: tests/unit/test_data_plane.py (hunk 2 above)
Comment (igiloh-pinecone) [reply tests/unit/test_data_plane.py:447]: This should probably be refactored so it uses its own pre-determined data, then this won't be necessary.

@rajat08 can you please take a look? I'm less familiar with this test case

File: tests/unit/test_control_plane.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jhamon) [reply tests/unit/test_control_plane.py:13]: Got it, makes sense. Thanks for the context.
